src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/TagHelperSpanVisitor.cs

```diff
@@ -0,0 +1,183 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Components;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+using Microsoft.VisualStudio.Editor.Razor;
+using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic
+{
+    internal class TagHelperSpanVisitor : SyntaxWalker
```

Comment:
Should rename this to be specific to semanticness. I know it's in the Semantic namespace but VS doesn't care about that when doing ctrl + comma for searching 😄. `TagHelperSemanticSpanVisitor`